include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
+      {
+        enum kind
+        {
+          continuous_fem_field = 0,
+          discontinuous_fem_field = 1,
+          particles = 2
```

Comment:
Only one `fem_field` was my first implementation, but with the distinction we can allow for different fields to have different finite element types (which was on our long-term to-do list anyway). This enum could also then supersede the `use_discontinuous_composition_discretization` variable that does not make sense with other advection methods like particles. This is not fully implemented in this PR, because I wanted to discuss it first.